src/daft-core/src/array/ops/utf8.rs

```diff
@@ -62,6 +62,17 @@ where
     ))
 }
 
+fn right_most_chars(val: &str, nchar: usize) -> String {
+    let len = val.chars().count();
+    if nchar == 0 || len == 0 {
+        String::new()
+    } else if nchar >= len {
+        val.to_string()
+    } else {
+        val.chars().skip(len - nchar).collect::<String>()
+    }
```

Comment:
```suggestion
fn right_most_chars(val: &str, nchar: usize) -> &str {
    let len = val.chars().count();
    if nchar == 0 || len == 0 {
        ""
    } else if nchar >= len {
        val
    } else {
        let skip = len.saturating_sub(nchar);
        val.char_indices().nth(skip).map_or("", |(i, _)| &val[i..])
    }
```

nit: use char_indices and return a string slice to avoid extra memory allocation